src/core/server/elasticsearch/client/cluster_client.ts

```diff
@@ -110,4 +116,43 @@ export class ClusterClient implements ICustomClusterClient {
       ...scopedHeaders,
     };
   }
+
+  private getScopedAuditor(request: ScopeableRequest) {
+    // TODO: support alternative credential owners from outside of Request context in #39430
+    if (isRealRequest(request)) {
+      const kibanaRequest =
+        request instanceof KibanaRequest ? request : KibanaRequest.from(request);
+      return this.auditorFactory.asScoped(kibanaRequest);
+    }
+  }
+
+  /** All requests made by clients are written in the auditor for further analysis. */
+  private integrateAuditor(
+    internalClient: ElasticsearchClient,
+    scopedClient: ElasticsearchClient,
+    request: ScopeableRequest,
+    id: string
+  ) {
+    const auditor = this.getScopedAuditor(request);
+    if (auditor) {
+      internalClient.on('request', (err: ApiError, event: RequestEvent) => {
```

Comment:
@delvedor I didn't expect that client children share event bus, but according to this test is desired behaviour. https://github.com/elastic/elasticsearch-js/blob/1a7968aba3bbbd9660d582db2252eeafa4b23260/test/unit/child.test.js#L238-L267

Reply:
Correct, the event emitter is the same for all the children.
You can easily understand "which client" emitted the event by using the [`name` configuration option](https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/observability.html#_client_name), which is present in the event metadata.

Reply:
I'm not sure it's expected behavior tbh. Let me check that the current code doesn't lead to a memory leak.

Reply:
I don't see how we can manage the `child` client lifecycle reliably since they can be created in plugin code at any time via `asScoped`.